oeps/architectural-decisions/oep-0065-frontend-composability.rst

```diff
@@ -0,0 +1,492 @@
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-65 <oep-0065-arch-frontend-composability>`
+   * - Title
+     - Frontend Composability
+   * - Last Modified
+     - 2024-04-12
+   * - Authors
+     -
+       * Adolfo R. Brandes <adolfo@axim.org>
+       * Pedro Martello <pedro@hammerlabs.net>
+       * David Joy <david@joy.engineering>
+   * - Arbiter
+     - Adam Stankiewicz <astankiewicz@2u.com>
+   * - Status
+     - Under Review
+   * - Type
+     - Architecture
+   * - Created
+     - 2024-04-03
+   * - Review Period
+     - April 15, 2024 - April 29, 2024
+   * - Resolution
+     - <links to any discussions where the final status was decided>
+   * - References
+     -
+       * `FC-0054 - Composable Micro-frontends Discovery <https://openedx.atlassian.net/wiki/spaces/COMM/pages/4063821827/FC-0054+-+Composable+Micro-frontends+Piral+Discovery>`_
+       * `FC-0007 - Modular MFE Domains Discovery <https://openedx.atlassian.net/wiki/spaces/COMM/pages/3614900241/CLOSED+FC-0007+-+Modular+MFE+Domains+Discovery>`_
+
+
+Abstract
+********
+
+This OEP proposes that Open edX micro-frontends (MFEs) adopt module federation as an approach to address inconsistencies, performance issues, and inflexibilities inherent in the current MFE architecture.
+
+Motivation
+**********
+
+Micro-frontends were originally designed to avoid some of the limitations of the monolithic ``edx-platform`` frontend, namely that otherwise independent teams were beholden to the build, test, and release lifecycle of the rest of the codebase. This dramatically slowed down the pace of frontend feature development, experimentation, and innovation.
+
+As a result, the Open edX "micro-frontend" architecture focused on creating an unopinionated, loosely coupled set of applications with the goal of enabling teams to iterate quickly and independently. This goal was successful.
+
+However, now roughly 6 years into this gradual re-platforming, we've discovered that the completely siloed nature of these MFEs has created its own set of problems.  Our frontends are more like independent single-page applications than micro-frontends, as we never invested in ways of composing them together. The problems inherent in today's MFE architecture are described below.
+
+Consistency
+===========
+
+Because each MFE is completely independent, this can create inconsistencies from MFE to MFE.
+
+Reusable components (Paragon)
+-----------------------------
+
+MFEs may use different versions of ``@openedx/paragon``, resulting in functional and stylistic differences.
+
+Header/Footer
+-------------
+
+MFEs may use different versions of ``@edx/frontend-component-header`` and ``@edx/frontend-component-footer``, also resulting in functional, stylistic, and content/navigation differences.
+
+Branding
+--------
+
+Brand packages created from ``@openedx/brand-openedx`` may be different versions, resulting in any number of subtle visual differences.
+
+Other Dependencies
```

Comment:
I don't think it is due to paragon or another dependency, but a visual difference that I notice a lot in different MFEs is the width of the page. If that could be solved through this effort it would also aid in consistency of the whole frontend.

Reply:
Great point - and I think having a shell app helps with this, which would be great to mention in here!  Going to work that into the section.